### PR TITLE
Allow for modification of Git Branch and Nginx HTTP Ports

### PR DESCRIPTION
**Pull request by MichaelObi**
Awesome stuff!
These are some modifications I made to make it easier for me to run the playbook. I think it'll help someone else too.

- Allows`git_branch` to be set in  `group_vars/all`. The set branch is checked out after cloning the repository.

- Allows `nginx_http_port` and `nginx_https_port` to be set in `group_vars/all` to modify the port Nginx should run on. This only works if `nginx_hostless_global_config` is set to `yes` in `group_vars/all`.

Cheers

### CODE REVIEW

File: roles/setup_venv_app/tasks/main.yml [hunk 1]
```diff
@@ -8,7 +8,10 @@
   command: "{{ virtualenv_cmd }} '{{ base_dir }}'"
 
 - name: Clone Git repository
-  command: "git clone {{ git_repo }} '{{ base_dir }}/appdata'"
+  git:
+    repo: https://emergingplatforms.visualstudio.com/RoboAdvisor/_git/RoboAdvisorBot
```
Comment (Kwpolska) [inline roles/setup_venv_app/tasks/main.yml]: I think you leaked your personal repo name here. That should still be `'{{ git_repo }}'`

File: group_vars/all [hunk 2]
```diff
@@ -12,6 +14,8 @@ uwsgi_env: []
 # uwsgi_env:
 # - 'DJANGO_SETTINGS_MODULE={{ app_package }}.settings'
 git_repo: https://github.com/Kwpolska/flask-demo-app
+# Set git branch name
+git_branch: master
```
Comment (Kwpolska) [inline group_vars/all]: :+1: (but that comment is unnecessary IMO)

File: group_vars/all [hunk 3]
```diff
@@ -4,6 +4,8 @@ app_package: flaskapp
 app_callable: app
 base_dir: '/srv/{{ app_name }}'
 nginx_server_name: 'localhost {{ app_name }}.local'
+nginx_http_port: 80
```
Comment (Kwpolska) [inline group_vars/all]: Is this configuration necessary? `nginx_https_port` is used in a comment that needs to be manually handled, so it’s better to leave it hard-coded. What use-case do you have for a different HTTP port than 80 though? For development, the built-in server of your framework will be better. For real-life deployment, most people want the standard port 80.

(I will accept it if you convince me)

**Review by Kwpolska — CHANGES_REQUESTED** [review]

File: roles/setup_venv_app/tasks/main.yml (hunk 1 above)
Comment (MichaelObi) [reply roles/setup_venv_app/tasks/main.yml]: ouch! Thanks for pointing this out

File: group_vars/all (hunk 2 above)
Comment (MichaelObi) [reply group_vars/all]: True!

File: group_vars/all (hunk 3 above)
Comment (MichaelObi) [reply group_vars/all]: In my case, I had to deploy on an in-house testing server that had Apache Tomcat already bound to port 80. I'd assume  a few people would run into the same situation. 
Ports 80 and 443 are set as safe defaults just incase no one needs it.

Comment (Kwpolska) [reply group_vars/all]: I’ll accept it for port 80 and drop it for port 443. I’ll also clean up the commits and fix git_repo specification.

Comment (MichaelObi) [reply group_vars/all]: Thanks!

**Review by Kwpolska — APPROVED** [review]

**Comment by Kwpolska** [comment]
Thanks for fixing this! I cleaned it up in your fork with a force-push, so you might need some extra work to get your fork up-to-date (if you’ll need it in the future).

**Comment by Kwpolska** [comment]
How should I add you to the AUTHORS file?

**Comment by MichaelObi** [comment]
Nothing would make me happier!

**Pull request merged**